service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -185,14 +185,32 @@ func (d *Decoder) decodeBinary(b []byte, v reflect.Value) error {
 		return nil
 	}
 
-	switch v.Interface().(type) {
-	case []byte:
+	if v.Kind() != reflect.Slice {
```

Comment:
What is this if statement checking? If it is a `[]byte` slice already, do we need to check that it is a slice?